vmware-tanzu/nsx-operator
# Add pod controller and node controller

1. Add the pod controller to watch pod events and create/update/delete NSX subnet
   port.
2. Add the node controller to cache the host transport node info and provided
   the context id for port creation.

Tests done:

1. Create a pod and check the NSX subnet port creation and pod annotations.
2. Create multiple pods to ensure that there're multiple subnets being created.
3. Force delete the pod during operator downtime and ensure the related NSX
   subnet port is deleted after the operator restarts.
3. Observe the logs to ensure the worker nodes are cached.
